grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/HttpStreamReader.java

```diff
@@ -181,7 +181,15 @@ public Void apply(@Nullable Void unused, @Nullable Throwable cause) {
         }
 
         if (cause == null) {
-            closeDeframer();
+            if (!deframer.isStalled()) {
```

Comment:
Ah I see - instead of this how about adding a `isClosing` method that checks `closeWhenComplete` and check it inside `closeDeframer` instead of doing this check. There shouldn't be any need to call `closeWhenComplete` twice.

Reply:
And probably a good idea to move this bug fix to a separate PR.

Reply:
> And probably a good idea to move this bug fix to a separate PR.

Yes, I'm preparing a PR to resolve this issue. But some tests failed, my change seems to cause side effects.
Let me elaborate on it after more investigating. 🤔